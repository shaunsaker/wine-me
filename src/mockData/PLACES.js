import PLACE from './PLACE';

const PLACES = { 1: PLACE };

export default PLACES;
