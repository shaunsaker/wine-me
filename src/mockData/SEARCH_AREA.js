const SEARCH_AREA = {
  name: 'Robertson',
  location: {
    lat: -33.9263749,
    lng: 18.9319848,
  },
  radius: 10,
  id: '1',
};

export default SEARCH_AREA;
