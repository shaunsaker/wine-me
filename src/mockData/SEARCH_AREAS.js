import SEARCH_AREA from './SEARCH_AREA';

const SEARCH_AREAS = { 1: SEARCH_AREA };

export default SEARCH_AREAS;
