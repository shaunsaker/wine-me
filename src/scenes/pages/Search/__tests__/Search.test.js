import React from 'react';
import renderer from 'react-test-renderer';

import { Search } from '..';
import SEARCH_AREAS from '../../../../mockData/SEARCH_AREAS';
import SEARCH_AREA from '../../../../mockData/SEARCH_AREA';
import PLACES from '../../../../mockData/PLACES';

jest.mock('../../../../components/PlaceList', () => 'PlaceList');

// Fixes _bezier is not a function bug
jest.useFakeTimers();

describe('Search', () => {
  const spies = [];
  const dispatch = jest.fn();
  const searchTerm = 'Water';
  const searchArea = SEARCH_AREA;

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);

      expect(component).toMatchSnapshot();
    });

    it('renders with a selected searchArea', () => {});

    it('renders with a searchTerm', () => {});
  });

  describe('methods', () => {
    it('should handle onBack', () => {
      spies[0] = jest.spyOn(Search.prototype, 'navigate');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.onBack();

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should handle onChangeText', () => {
      spies[0] = jest.spyOn(Search.prototype, 'setSearchTerm');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.onChangeText(searchTerm);

      expect(spies[0]).toHaveBeenCalledWith(searchTerm);
    });

    it('should handle onSubmit', () => {
      spies[0] = jest.spyOn(Search.prototype, 'dismissKeyboard');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.onSubmit();

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should handle onSearchAreaLabelPress', () => {
      spies[0] = jest.spyOn(Search.prototype, 'setSearchArea');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.onSearchAreaLabelPress(searchArea);

      expect(spies[0]).toHaveBeenCalledWith(searchArea);
    });

    it('should handle onSelectedSearchAreaLabelPress', () => {
      spies[0] = jest.spyOn(Search.prototype, 'setSearchArea');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.onSelectedSearchAreaLabelPress();

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should handle setSearchTerm', () => {
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.setSearchTerm(searchTerm);

      expect(instance.state.searchTerm).toEqual(searchTerm);
    });

    it('should handle dismissKeyboard', () => {
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.dismissKeyboard();
    });

    it('should handle setSearchArea', () => {
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.setSearchArea(searchArea);

      expect(instance.state.searchArea).toEqual(searchArea);
    });

    it('should handle navigate', () => {
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const instance = component.getInstance();

      instance.navigate();
    });
  });

  describe('actions', () => {
    it('should call onBack on back button press', () => {
      spies[0] = jest.spyOn(Search.prototype, 'navigate');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const { root } = component;
      const targetComponent = root.findByProps({ testID: 'search.button.back' });

      targetComponent.props.handlePress();

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should call onSelectedSearchAreaLabelPress on selected label press', () => {
      spies[0] = jest.spyOn(Search.prototype, 'onSelectedSearchAreaLabelPress');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const { root } = component;

      // Setup
      const instance = component.getInstance();
      instance.setSearchArea(searchArea);

      const targetComponent = root.findByProps({
        testID: `search.label.selected.${searchArea.name}`,
      });

      targetComponent.props.handlePress();

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should call onChangeText on input change text', () => {
      spies[0] = jest.spyOn(Search.prototype, 'onChangeText');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const { root } = component;
      const targetComponent = root.findByProps({ testID: 'search.input' });

      targetComponent.props.onChangeText(searchTerm);

      expect(spies[0]).toHaveBeenCalledWith(searchTerm);
    });

    it('should call onSubmit on input submit', () => {
      spies[0] = jest.spyOn(Search.prototype, 'onSubmit');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const { root } = component;
      const targetComponent = root.findByProps({ testID: 'search.input' });

      targetComponent.props.onSubmitEditing();

      expect(spies[0]).toHaveBeenCalled();
    });

    it('should call onSearchAreaLabelPress on unselected label press', () => {
      spies[0] = jest.spyOn(Search.prototype, 'onSearchAreaLabelPress');
      const component = renderer.create(<Search searchAreas={SEARCH_AREAS} places={PLACES} />);
      const { root } = component;
      const targetComponent = root.findByProps({ testID: `search.label.${searchArea.name}` });

      targetComponent.props.handlePress();

      expect(spies[0]).toHaveBeenCalledWith(searchArea);
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
