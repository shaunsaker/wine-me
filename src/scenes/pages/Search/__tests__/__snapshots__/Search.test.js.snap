// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders renders with minimum required props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "paddingTop": 22,
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#48466D",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#48466D",
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingRight": 15,
            "paddingVertical": 10,
          }
        }
        testID="search.button.back"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#FFFFFF",
                "fontSize": 28,
                "marginHorizontal": -9,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgba(0, 0, 0, 0.70)",
            "borderRadius": 5,
            "flex": 1,
            "flexDirection": "row",
            "height": 48,
            "opacity": 1,
            "paddingHorizontal": 15,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontSize": 12,
              },
              Object {
                "color": "#FFFFFF",
                "fontSize": 24,
                "marginRight": 7.5,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          autoFocus={true}
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="Search"
          placeholderTextColor="#FFFFFF"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Nunito-Regular",
                "fontSize": 17,
              },
            ]
          }
          testID="search.input"
          underlineColorAndroid="transparent"
          value={null}
        />
      </View>
    </View>
    <View
      style={
        Object {
          "paddingTop": 10,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={Object {}}
        data={
          Array [
            Object {
              "id": "1",
              "location": Object {
                "lat": -33.9263749,
                "lng": 18.9319848,
              },
              "name": "Robertson",
              "radius": 10,
            },
          ]
        }
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={Object {}}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View>
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "flexDirection": "row",
                },
                null,
              ]
            }
          >
            <View
              style={
                Object {
                  "marginRight": 7.5,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "backgroundColor": "#E23E57",
                    "borderColor": null,
                    "borderRadius": 5,
                    "borderWidth": 0,
                    "elevation": 2,
                    "opacity": 1,
                    "paddingHorizontal": 7.5,
                    "paddingVertical": 5,
                    "shadowColor": "#212121",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": 0,
                    },
                    "shadowOpacity": 0.17,
                    "shadowRadius": 2,
                  }
                }
                testID="search.label.Robertson"
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Nunito-SemiBold",
                      "fontSize": 11,
                    }
                  }
                >
                  Robertson
                </Text>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "flex": 1,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Object {
          "flex": 1,
        }
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "flex": 1,
              "justifyContent": "center",
              "maxWidth": 300,
              "paddingVertical": 100,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#3D84A8",
                  "fontSize": 64,
                  "marginBottom": 10,
                },
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#212121",
                "fontFamily": "Nunito-SemiBold",
                "fontSize": 24,
                "marginBottom": 10,
                "textAlign": "center",
              }
            }
          >
            Search over 525+ places in the Western Cape
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#212121",
                "fontFamily": "Nunito-Regular",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            If you're heading somewhere specific, try an area search by tapping on of the red buttons.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
